plugins/metadata/MetadataPlugin.php

```diff
@@ -46,6 +46,13 @@ public static function updateMetadataFileSync(Metadata $dbMetadata, $xmlData)
 		$dbMetadata->incrementVersion();
 		$key = $dbMetadata->getSyncKey(Metadata::FILE_SYNC_METADATA_DATA);
 		kFileSyncUtils::file_put_contents($key, $xmlData);
+		Propel::disableInstancePooling();
+		$currentMetadata = MetadataPeer::retrieveByPK($dbMetadata->getId());
+		Propel::enableInstancePooling();
+		if ($currentMetadata->getVersion() > $dbMetadata->getVersion())
+		{
+			throw new kCoreException("Metadata version already increased by a different process", kCoreException::INTERNAL_SERVER_ERROR);
```

Comment:
need to make sure that this core exception is translate into api exception

Reply:
Changed to a custom KalturaAPIException.